internal/configs/node/config.go

```diff
@@ -218,8 +219,19 @@ func SetNetworkType(networkType NetworkType) {
 }
 
 // SetArchival set archival mode
-func (conf *ConfigType) SetArchival(archival bool) {
-	defaultConfig.isArchival = archival
+// for beacon chain node, the archival variable will
+// overrdie bcArchival as the shardID is the same
+func (conf *ConfigType) SetArchival(bcArchival, archival bool) {
+	if conf.isArchival == nil {
+		conf.isArchival = make(map[uint32]bool)
+	}
+	conf.isArchival[shard.BeaconChainShardID] = bcArchival
+	conf.isArchival[conf.ShardID] = archival
+}
+
+// IsArchival return the isArchival map
+func (conf *ConfigType) IsArchival() map[uint32]bool {
```

Comment:
suggest rename isArchival (usually a bool) to ArchivalModes or ArchivalSettings etc.